content/understand-django/2020-01-22-urls-lead-way.es.md

```diff
@@ -0,0 +1,444 @@
+{{< extlink "https://docs.python.org/3/howto/regex.html" "Regular Expression HOWTO" >}}
+de Python.
+
+Saber que este poder con `re_path` está ahí puede ayudarte más adelante en tu camino con Django, incluso si no lo necesitas ahora.
+
+## Agrupación de URL relacionadas
+
+Hasta este punto, hemos analizado rutas individuales que puede mapear en una URLconf. ¿Qué podemos hacer cuando un grupo relacionado de puntos de vista debe compartir un camino común? ¿Por qué querríamos hacer esto?
+
+Imaginemos que estás construyendo un proyecto educativo. En su proyecto, tiene escuelas, estudiantes y otros conceptos relacionados con la educación. Podrías hacer algo como:
+
+```python
+# project/urls.py
+from django.urls import path
+
+from schools import (
+    views as schools_views,
+)
+from students import (
+    views as students_views,
+)
+
+urlpatterns = [
+    path(
+        "schools/", schools_views.index
+    ),
+    path(
+        "schools/<int:school_id>/",
+        schools_views.school_detail,
+    ),
+    path(
+        "students/",
+        students_views.index,
+    ),
+    path(
+        "students/<int:student_id>/",
+        students_views.student_detail,
+    ),
+]
+```
+
+Este enfoque funciona bien, pero obliga a la raíz URLconf a conocer todas las vistas definidas en cada aplicación, `schools` y `students`. En cambio, podemos usar `include` para manejar esto mejor.
+
+```python
+# project/urls.py
+from django.urls import include, path
+
+urlpatterns = [
+    path(
+        "schools/",
+        include("schools.urls"),
+    ),
+    path(
+        "students/",
+        include("students.urls"),
+    ),
+]
+```
+
+Entonces, en cada aplicación, tendríamos algo como:
+
+```python
+# schools/urls.py
+from django.urls import path
+
+from schools import views
+
+urlpatterns = [
+    path("", views.index),
+    path(
+        "<int:school_id>/",
+        views.school_detail
+    ),
+]
+```
+
+El uso de `include` le da a cada aplicación de Django autonomía en qué vistas necesita definir. El proyecto puede ser felizmente "ignorante" de lo que está haciendo la aplicación.
+
+Adicionalmente, se elimina la repetición de `schools/` o `students/` del primer ejemplo. A medida que Django procesa una ruta, coincidirá con la primera parte de la ruta y pasará el *resto* a la URLconf que se define en la aplicación individual. De esta forma, las configuraciones de URL pueden formar un árbol donde la raíz URLconf es donde comienzan todas las solicitudes, pero las aplicaciones individuales pueden manejar los detalles a medida que una solicitud se enruta a la aplicación adecuada.
+
+## Nombrar URLs
+
+Hemos analizado las principales formas en que las URL se definen con `path`, `re_path`, e `include`. Hay otro aspecto a considerar. ¿Cómo podemos referirnos a las URL en otros lugares del código? Considere esta vista (bastante tonta):
```

Comment:
Was this a mistake? I don't know what `e` means by itself in Spanish.
```suggestion
Hemos analizado las principales formas en que las URL se definen con `path`, `re_path`, y `include`. Hay otro aspecto a considerar. ¿Cómo podemos referirnos a las URL en otros lugares del código? Considere esta vista (bastante tonta):
```

Reply:
Neat! Thanks for educating me on this. I took a few years of Spanish in high school and never encountered this rule. It's fascinating. I guess it's similar to how English uses `an` in front of a word that starts with a vowel sound.

Reply:
Exactly!